docker_auth.go

```diff
@@ -81,6 +81,19 @@ func getDockerAuthConfigs() (map[string]types.AuthConfig, error) {
 		cfgs[k] = ac
 	}
 
+	// in the case where the auth field in the .docker/conf.json is empty, and the user has credential helpers registered
+	// the auth comes from there
+	if len(cfg.AuthConfigs) == 0 && len(cfg.CredentialHelpers) > 0 {
```

Comment:
I think this check is simpler now, as the body of the inner for/range loop will not be evaluated
```suggestion
	if len(cfg.AuthConfigs) == 0 {
```